### ISSUE TITLE
Disable ID names that Oxygen automatically assigns

### ISSUE THREAD

**Issue by dnnsjsk** [soflyy/oxygen-bugs-and-features#14]
**Describe the feature you'd like to see included in Oxygen.**
*A clear, detailed description of the feature you're requesting.*

I would love a "Developer Mode", so Oxygen wouldn't name all of the div IDs i am creating.

**What are the use cases for this feature?**
*e.g. Accessibility, e-commerce store that sells bicycles, etc.*

Clean code. I am using BEM naming methodology for all my CSS classes, thus already creating a clear readable structure.. all those random ID names are kinda obscuring it. see screenshot.

![classes](https://user-images.githubusercontent.com/39319746/51288203-5ed4c980-19fb-11e9-9c40-b306efdb8672.png)

**Comment by wplit** [comment]
+1 for developer mode. The ability to edit the html further so we can add our own attributes would be ideal for this mode also.

**Comment by LordWhisper** [comment]
> +1 for developer mode. The ability to edit the html further so we can add our own attributes would be ideal for this mode also.

+1 

This would actually be god mode for Oxygen and make it even more superior to all other builders.

**Comment by LordWhisper** [comment]
I think the problem is, it needs to have an ID at least at the very moment you do some kind of changes - before that, there is no need for it, you're right with that. 

@dnnsjsk : You can rename those IDs to your liking already, did you know that? Just click them again when it asks you to add the class name. Change it and boom....

**Comment by Gibsonion** [comment]
I do like this idea, but in the meantime if you weren't aware it is possible to edit IDs to custom ones to make it more semantic.

**Comment by EmSixTeen** [comment]
Please god. This is driving me mental.

**Comment by schietegal** [comment]
+1 Yes Please! Give us the option to "disable" the IDs.

**Comment by blaci29** [comment]
+1 Give us the option to add identifiers only for the divs we need.

**Comment by wpsumo** [comment]
Yes please, there is no reason to add these IDs. Threat IDs as we do with classes by adding it once the user want to add a unique styling identifier. Bricks builder is removing and cleaning things up and wouldn't surprise if they remove IDs as well as there are around 300 upvotes on their board:
https://www.facebook.com/groups/brickscommunity/posts/901471573850697/
https://bricksbuilder.io/ideas/#3183

I think most users prefers it to be cleaner. I hope to see the ID element as an option exactly how classes are treated, and not as a standard implementation.
**Keep the DOM clean and as small as possible.**


Duplicate IDs is a problem but "solved" by moving it to attributes. Just solved the duplicate ID issue, not cleaning things up. Duplicate ID is only solved in repeater in Oxygen 4.0 beta, as Oxygen add static IDs to all element we end up with duplicate ID issues on the gutenberg blocks as well if the same block is used more than once on a page as expressed here:
https://github.com/soflyy/oxygen-bugs-and-features/issues/432

**Comment by gerogr** [comment]
+1

**Comment by IKuipers1** [comment]
+1

**Comment by EmSixTeen** [comment]
Is there a reason this isn't fixed?

**Comment by IKuipers1** [comment]
This was their answer:

I'm very sorry for the inconvenience. Our development is aware of the
issue, but due to the complexity of the issue, there's currently no fix for
it.

This is an issue that our development team is aware of and investigating.
Unfortunately, I cannot provide further details as to when it may be
addressed.

Op za 23 apr. 2022 om 11:03 schreef Aaron ***@***.***>:

**Comment by IKuipers1** [comment]
I asked this a week ago. This was his answer:

>> Why do we need an ID on every element? Couldn't it be removed (on e.g. div, span, h, a, p, img ) when no style is added?


Oxygen creates an ID for every element. This helps the program understand how to render content on the front end. I'm very sorry, but there's no built-in way to remove the ID of an element created by Oxygen. If that is something you'd like to see in a future update. I'd encourage you to upvote a similar feature request here: https://github.com/soflyy/oxygen-bugs-and-features/issues/14.

**Comment by LauraHamburg** [comment]
+10

**Comment by Ikpixel** [comment]
This will make Oxygen Builder better then it already is.

**Comment by ElliottBear** [comment]
This is absolutely essential for Oxygen to be the perfect web editor - just having a developer mode available in the site settings, and then have the primary and advanced tab for an element greyed out until the user adds either a class or an ID would be perfect
